ci/scripts/build.sh

```diff
@@ -61,27 +59,38 @@ done
 
 pushd "${build_dir}"
 
-if [ "${ARROW_JAVA_SHADE_FLATBUFFERS}" == "ON" ]; then
+# TODO: ARROW_JAVA_SHADE_FLATBUFFERS isn't used for our artifacts. Do
+# we need this?
+# See also: https://github.com/apache/arrow/issues/22021
+if [ "${ARROW_JAVA_SHADE_FLATBUFFERS:-OFF}" == "ON" ]; then
```

Comment:
I think we need to revisit this sometime because of https://github.com/apache/arrow-java/issues/67 (and it appears Google has closed the issue asking them to fix it upstream). 

AFAIK the discussion in the issue you found is weird...We shouldn't need to change imports ourselves, the shading should take care of that. But that's not relevant here.

Reply:
OK. Can we use #67 for this topic? Or should we create a separated issue for this topic?

Reply:
Yes, we can use #67!